docs/modules/ROOT/pages/design/database-migrations.adoc

```diff
@@ -0,0 +1,56 @@
+= Database Migrations
+
+This document details the database migration strategy KubeArchive uses. The main goals
+of the migrations are:
+
+* Make the upgrade experience easier
+* Grant the ability of undo DB changes if something goes wrong
+
+**Note**: the implementation of database migrations do not include the execution of those
+migrations. We believe that production databases should not be modified in an unattended way
+so we leave to the KubeArchive operator the migration executions.
+
+== Requirements
+
+* Migrations are run by the PostgreSQL user that KubeArchive uses, to avoid permission problems.
+* Migrations migrate data as well as the schema.
+* Migrations have forward and backwards implementation (`up` and `down` files).
+
+== Implementation
+
+We decided to use the
+link:https://github.com/golang-migrate/migrate[`migrate` tool]
+due to its popularity and because its maintained. This tool automates certain
+processes related with migrations, but its a thin layer that can be easily understood
+and replaced if needed.
```

Comment:
```suggestion
We decided to use the
link:https://github.com/golang-migrate/migrate[`migrate` tool]
due to its popularity and because it's well maintained. It automates certain
processes related with migrations, but it's a thin layer that can be easily understood
and replaced if needed.
```